### PR TITLE
roachtest: add COCKROACH_UPGRADE_TO_DEV_VERSION to multitenant-upgrade

### LINKED ISSUE #88345: roachtest: multitenant-upgrade failed
roachtest.multitenant-upgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6549169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6549169?buildTab=artifacts#/multitenant-upgrade) on master @ [bbcb10f982e47943de63a3a2d1faf10d3eb49e75](https://github.com/cockroachdb/cockroach/commits/bbcb10f982e47943de63a3a2d1faf10d3eb49e75):

```
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString

	cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:259,cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:259,cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:265,sql_runner.go:176,sql_runner.go:205,multitenant_upgrade.go:314,multitenant_upgrade.go:162,multitenant_upgrade.go:35,test_runner.go:928: cluster.RunE: context canceled
		(1) attached stack trace
		  -- stack trace:
		  | main.(*clusterImpl).RunE
		  | 	main/pkg/cmd/roachtest/cluster.go:1950
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1931
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatal
		  | 	main/pkg/cmd/roachtest/test_impl.go:259
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1933
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatal
		  | 	main/pkg/cmd/roachtest/test_impl.go:259
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1933
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatalf
		  | 	main/pkg/cmd/roachtest/test_impl.go:265
		  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).Query
		  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:176
		  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).QueryStr
		  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:205
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.verifySQL
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:314
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runMultiTenantUpgrade
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:162
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerMultiTenantUpgrade.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:35
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:928
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1594
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/multi-tenant
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*multitenant-upgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-19779

### PR DESCRIPTION
**Pull request by dt**
fixes https://github.com/cockroachdb/cockroach/issues/88345

Release note: none.

### CODE REVIEW

File: pkg/cmd/roachtest/tests/multitenant_upgrade.go [hunk 1]
```diff
@@ -72,6 +72,7 @@ func runMultiTenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster,
 	kvNodes := c.Node(1)
 
 	settings := install.MakeClusterSettings(install.BinaryOption(predecessorBinary), install.SecureOption(true))
+	settings.Env = append(settings.Env, "COCKROACH_UPGRADE_TO_DEV_VERSION=true")
```
Comment (rafiss) [inline pkg/cmd/roachtest/tests/multitenant_upgrade.go:75]: lgtm, but we could also append everything in `config.DefaultEnvVars()` here while we're at it

**Review by rafiss — APPROVED** [review]

**Review by ajstorm — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88503#-:-NCekvBA76vp9HNX4F40:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @rafiss)

File: pkg/cmd/roachtest/tests/multitenant_upgrade.go (hunk 1 above)
Comment (dt) [reply pkg/cmd/roachtest/tests/multitenant_upgrade.go:75]: I think that's already there?
The line above is `install.MakeClusterSettings` which looks like it consists of `Env:           config.DefaultEnvVars(),`

Comment (dt) [reply pkg/cmd/roachtest/tests/multitenant_upgrade.go:75]: ... which means `COCKROACH_UPGRADE_TO_DEV_VERSION` is already in the env. Okay, I have no idea why I am seeing the test pass now but not before. I'm guessing I have a stale binary or something somewhere.

Comment (rafiss) [reply pkg/cmd/roachtest/tests/multitenant_upgrade.go:75]: yeah i don't see either of these lines referencing `settings`:
```
	tenant11 := createTenantNode(ctx, t, c, kvNodes, tenant11ID, tenantNode, tenant11HTTPPort, tenant11SQLPort, tenantStartOpt)
	tenant11.start(ctx, t, c, predecessorBinary)
```

File: pkg/cmd/roachtest/tests/multitenant_utils.go [hunk 2]
```diff
@@ -250,7 +250,7 @@ func startTenantServer(
 	errCh := make(chan error, 1)
 	go func() {
 		errCh <- c.RunE(tenantCtx, node,
-			append([]string{binary, "mt", "start-sql"}, args...)...,
+			append([]string{"COCKROACH_UPGRADE_TO_DEV_VERSION=true", binary, "mt", "start-sql"}, args...)...,
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/cmd/roachtest/tests/multitenant_utils.go]: done

**Review by rafiss — APPROVED** [review]

**Comment by rafiss** [comment]
i'll merge this to resolve the failure before the release triage meeting

bors r+

**Pull request merged**